controllers/mongodb_tls.go

```diff
@@ -91,23 +73,35 @@ func (r *ReplicaSetReconciler) validateTLSConfig(mdb mdbv1.MongoDBCommunity) (bo
 	// Watch certificate-key secret to handle rotations
 	r.secretWatcher.Watch(mdb.TLSSecretNamespacedName(), mdb.NamespacedName())
 
+	// Watch CA certificate changes
+	if mdb.Spec.Security.TLS.CaCertificateSecret != nil {
+		r.secretWatcher.Watch(mdb.TLSCaCertificateSecretNamespacedName(), mdb.NamespacedName())
+	} else {
+		r.configMapWatcher.Watch(mdb.TLSConfigMapNamespacedName(), mdb.NamespacedName())
+	}
+
```

Comment:
Q: Can both be specified, In this case we would like to prefer Secret over ConfigMap right?

Reply:
This was the current behaviour, yes, the secret takes precedence.